src/main/java/org/zalando/nakadi/controller/SubscriptionStreamController.java

```diff
@@ -135,6 +143,13 @@ public StreamingResponseBody streamEvents(
             SubscriptionStreamer streamer = null;
             final SubscriptionOutputImpl output = new SubscriptionOutputImpl(response, outputStream);
             try {
+                if  (subscriptionDbRepository.getSubscription(subscriptionId).getEventTypes().stream()
+                        .map(etName -> floodService.isConsumptionBlocked(etName)).findFirst().orElse(false)) {
+                    response.setStatus(HttpStatus.TOO_MANY_REQUESTS.value());
```

Comment:
This is a hard one. Fail closed means problems with the rate limiter will stop good writes from succeeding; fail open means bad actors could succeed in taking you and the rest of your customers out. As per `due to the last experiences on live as a fast solution` you could ship this now, but it's really really worth a think-through of the consequences pretty soon (a classic cascading failure mode is suppressing errors). So maybe :shipit: but revisit soon.